stripped/colabs/Analiza_Korelacji_NASA_stripped.py

```diff
@@ -0,0 +1,122 @@
+"""
+<a href="https://colab.research.google.com/github/PiotrMaciejKowalski/BigData2024Project/blob/CU-86bwcdr5f_Analiza-korelacji-rolinnoci-z-pozostaymi-zmiennymi_Piotr-Puchalski/colabs/Analiza_Korelacji_NASA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>
+"""
+
+"""
+# W tym pliku zajmiemy się analizą korealcji między roślinnością, a innymi cechami zawartymi w naszych danych
+"""
+
+!pip install pyspark
+
+
+
+import pyspark
+from pyspark.sql import SparkSession
+from pyspark.sql.types import IntegerType, FloatType, StringType, StructType
+spark = SparkSession.builder.appName('Corr_Analysis').getOrCreate()
+
+
+"""
+Teraz zaimportujemy dane
+"""
+
+from google.colab import drive
+import pandas as pd
+drive.mount('/content/drive')
+
+sampled = pd.read_csv('/content/drive/MyDrive/BigMess/NASA/sampled_NASA_200k.csv')
+
+schemat = StructType()
+for i in sampled.columns:
+  if i == "Date":
+    schemat = schemat.add(i, StringType(), True)
+  else:
+    schemat = schemat.add(i, FloatType(), True)
+
+nasa = spark.read.format('csv').option("header", True).schema(schemat).load('/content/drive/MyDrive/BigMess/NASA/NASA.csv')
+nasa.show(5)
+
+
+"""
+Po wczytaniu danych usuniemy kolumny, które nie będą nam potrzebne przy analizie korelacji roślinności z pozostałymi zmiennymi
+"""
+
+# usuwamy z danych kolumny odpowiadające za długość i szerokość geograficzną oraz datę
+columns_to_delete = ['lon', 'lat', 'Date']
+nasa = nasa.drop(*columns_to_delete)
+nasa.show(5)
+
+
+"""
+W naszych rozważaniach zajmiemy się głównie kolumną "GVEG", która odpowiada za wskaźnik "zielonej roślinności". Odnosi się do roślin o zielonych liściach, które są zdolne do fotosyntezy. Poniżej stowrzymy macierz korelacji, a następnie przedstawimy ją w postaci heat mapy.
+"""
+
+from pyspark.ml.stat import Correlation
+from pyspark.ml.feature import VectorAssembler
+
+# convert to vector column first
+vector_col = "corr_features"
+assembler = VectorAssembler(inputCols=nasa.columns, outputCol=vector_col, handleInvalid = 'skip')
+nasa_vector = assembler.transform(nasa).select(vector_col)
+
+# get correlation matrix
+corr_matrix = Correlation.corr(nasa_vector, vector_col)
+
+
+corr_matrix_pd = pd.DataFrame(np.array(corr_matrix.collect()[0][0].toArray()), columns=nasa.columns, index=nasa.columns)
+corr_matrix_np = corr_matrix_pd.to_numpy()
+
+
+# Tworzymy nasza heatmapę. Niestety funkcje pokroju sns.heatmap nie nadają się same w sobie do prezentacji tak dużych macierzy korelacji zatem potrzebne sa im pewne modyfikacje. Poniżej stowrzymy nasz wykres w taki sposób
+# aby po kliknięciu na grafikę ulegała ona przybliżeniu i wyostrzeniu
+
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+
+print(plt.get_backend())
+
+
+corr_matrix_np[np.triu_indices_from(corr_matrix_np)] = np.nan
```

Comment:
To powinno być zebrane jako jedna funkcja, nie kod-spaghetti

Reply:
Zrobione